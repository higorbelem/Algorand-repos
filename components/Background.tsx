import { StyleSheet, View, ViewProps } from "react-native"
import {Defs, Ellipse, RadialGradient, Stop, Svg} from 'react-native-svg';

import { Colors } from "@/constants/Colors";

export type BackgroundProps = ViewProps & {};

export const Background = ({style, children}: BackgroundProps) => {
    return(
        <View style={styles.container}>
            <Svg height="100%" width="100%" viewBox="0 0 100 100" style={styles.svg}>
                <Defs>
                    <RadialGradient
                        id="grad"
                        cx="-200" cy="-200" rx="400" ry="400"
                        fx="-200" fy="-200"
                        gradientUnits="userSpaceOnUse"
                    >
                        <Stop offset="0" stopColor={Colors.main} stopOpacity="1" />
                        <Stop offset="1" stopColor={Colors.background} stopOpacity="1" />
                    </RadialGradient>
                </Defs>
                <Ellipse cx="-200" cy="-200" rx="400" ry="400" fill="url(#grad)" />
            </Svg>

            <View style={[styles.content, style]}>
                {children}
            </View>
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        width: '100%',
        backgroundColor: Colors.background
    },
    content: {
        flex: 1,
        width: '100%'
    },
    svg: {
        position: 'absolute',
        width: '100%',
        height: '100%'
    }
});