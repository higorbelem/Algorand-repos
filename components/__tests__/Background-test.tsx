import * as React from 'react';
import { View } from 'react-native';
import { render } from '@testing-library/react-native';

import { Background } from '../Background';

describe('Background component', () => {
  it('Renders correctly', () => {
    const tree = render(<Background />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Should show children', async () => {
    const tree = render(<Background><View testID='children-view'/></Background>);
    
    const component = tree.getAllByTestId('children-view');

    expect(component).toBeTruthy();
  });
});