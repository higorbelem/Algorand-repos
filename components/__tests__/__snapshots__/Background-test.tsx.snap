// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Background component Renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#191C21",
      "flex": 1,
      "width": "100%",
    }
  }
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight="100%"
    bbWidth="100%"
    focusable={false}
    height="100%"
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "height": "100%",
          "position": "absolute",
          "width": "100%",
        },
        {
          "flex": 0,
          "height": "100%",
          "width": "100%",
        },
      ]
    }
    vbHeight={100}
    vbWidth={100}
    width="100%"
  >
    <RNSVGGroup
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
    >
      <RNSVGDefs>
        <RNSVGRadialGradient
          cx="-200"
          cy="-200"
          fx="-200"
          fy="-200"
          gradient={
            [
              0,
              -13816335,
              1,
              -15131615,
            ]
          }
          gradientTransform={null}
          gradientUnits={1}
          name="grad"
          rx="400"
          ry="400"
        />
      </RNSVGDefs>
      <RNSVGEllipse
        cx="-200"
        cy="-200"
        fill={
          {
            "brushRef": "grad",
            "type": 1,
          }
        }
        propList={
          [
            "fill",
          ]
        }
        rx="400"
        ry="400"
      />
    </RNSVGGroup>
  </RNSVGSvgView>
  <View
    style={
      [
        {
          "flex": 1,
          "width": "100%",
        },
        undefined,
      ]
    }
  />
</View>
`;
